### PR TITLE
Improve Node.getChildren() implementations

### PR DESCRIPTION
**Pull request by findepi**

### CODE REVIEW

File: presto-parser/src/main/java/com/facebook/presto/sql/tree/Select.java [hunk 1]
```diff
@@ -62,9 +62,9 @@ public boolean isDistinct()
     }
 
     @Override
-    public List<Node> getChildren()
+    public List<? extends Node> getChildren()
     {
-        return null;
+        return selectItems;
```
Comment (kokosing) [inline presto-parser/src/main/java/com/facebook/presto/sql/tree/Select.java]: can you check if there no other nodes with such implementation?

Comment (findepi) [inline presto-parser/src/main/java/com/facebook/presto/sql/tree/Select.java]: i didn't find any.

**Comment by findepi** [comment]
> Why is this needed?

@martint, if you ask about changing the signature of `getChildren()`, then -- not strictly needed. I could get away without this change, by using `ImmutableList.copyOf` in the implementations. Or using unchecked casts. Unchecked casts aren't pretty and `ImmutableList.copyOf` was flagged as "no need to do this" (first comment in `Select.java` under #7113 where this PR was extracted from)

**Comment by martint** [comment]
Sounds good, but please add an explanation to the commit message. Otherwise, it's not clear what relaxing the contract means or what it's for by looking at the change.

**Comment by findepi** [comment]
@martint sure, I put some explanation in the commit message too.

**Comment by findepi** [comment]
@martint  thanks for the review

**Pull request merged**